packages/ts/react-i18n/src/index.ts

```diff
@@ -150,43 +154,94 @@ export class I18n {
     this.#chunks.add(chunkName);
 
     if (this.#language.value) {
-      await this.updateLanguage(this.#language.value, false, chunkName);
+      await this.updateLanguage({ chunkName });
     }
   }
 
-  private async updateLanguage(newLanguage: string, updateSettings = false, newChunk?: string) {
-    if (this.#language.value === newLanguage && !newChunk) {
+  private async requestKeys(keys: readonly string[]): Promise<void> {
+    if (this.#batchedKeysPromise) {
+      // Keys request is being queued - add keys to the batch.
+      for (const k of keys) {
+        this.#batchedKeys.add(k);
+      }
+      return;
+    }
+
+    const nonBatchedKeys = keys.filter((k) => !this.#batchedKeys.has(k));
+    if (nonBatchedKeys.length === 0) {
+      // Keys request for these is already in progress - skip another request.
+      return;
+    }
+
+    // New request
+    this.#batchedKeys.clear();
+    for (const k of nonBatchedKeys) {
+      this.#batchedKeys.add(k);
+    }
+    // Wait to possibly collect other synchronously requested keys
+    this.#batchedKeysPromise = Promise.resolve(this.#batchedKeys);
+    await this.#batchedKeysPromise;
+    this.#batchedKeysPromise = undefined;
+    const batchedKeys = [...this.#batchedKeys];
+    return this.updateLanguage({ keys: batchedKeys }).then(() => {
+      console.warn(['A server call was made to translate keys those were not loaded:', ...batchedKeys].join('\n - '));
```

Comment:
Let's make tests pass:

```suggestion
      console.warn(['A server call was made to translate keys those were not loaded:', ...batchedKeys].join('\n  - '));
```